From 534e5b60130c08ca03e4dc6516e6c6f42cb63407 Mon Sep 17 00:00:00 2001
From: Thomas Schoebel-Theuer <schoebel@bell.site>
Date: Wed, 26 Jun 2013 14:46:41 +0200
Subject: [PATCH] mars: necessary for in-tree build

This patch should not go upstream (as long as the MARS source is
out-of-tree), because the kernel will no longer build without
MARS sources present in block/mars/ .

Use this for in-tree builds of the out-of-tree MARS sourcecode,
i.e. cd block/ && git clone https://github.com/schoebel/mars
---
 block/Kconfig  | 2 ++
 block/Makefile | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/block/Kconfig b/block/Kconfig
index e97934e..ac02d66 100644
--- a/block/Kconfig
+++ b/block/Kconfig
@@ -99,6 +99,8 @@ config BLK_DEV_THROTTLING
 
 	See Documentation/cgroups/blkio-controller.txt for more information.
 
+source block/mars/kernel/Kconfig
+
 endif # BLOCK
 
 config BLOCK_COMPAT
diff --git a/block/Makefile b/block/Makefile
index 514c6e4..32d37e2 100644
--- a/block/Makefile
+++ b/block/Makefile
@@ -17,3 +17,5 @@ obj-$(CONFIG_IOSCHED_CFQ)	+= cfq-iosched.o
 
 obj-$(CONFIG_BLOCK_COMPAT)	+= compat_ioctl.o
 obj-$(CONFIG_BLK_DEV_INTEGRITY)	+= blk-integrity.o
+
+obj-$(CONFIG_MARS)		+= mars/kernel/
-- 
1.8.1.4

